test/optimization/test_optimization_problem.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test OptimizationProblem """
+
+import os.path
+import tempfile
+from test.optimization.common import QiskitOptimizationTestCase
+import qiskit.optimization.problems.optimization_problem
+from qiskit.optimization.utils.qiskit_optimization_error import QiskitOptimizationError
+
+
+class TestOptimizationProblem(QiskitOptimizationTestCase):
+    """Test OptimizationProblem without the members that have separate test classes
+    (VariablesInterface, etc)."""
+
+    def setUp(self):
+        super().setUp()
+        self.resource_file = './test/optimization/resources/op_ip2.lp'
+
+    def test_constructor1(self):
+        op = qiskit.optimization.OptimizationProblem()
+        op.variables.add(names=['x1', 'x2', 'x3'])
+        self.assertEqual(op.variables.get_num(), 3)
+
+    def test_constructor2(self):
+        with self.assertRaises(QiskitOptimizationError):
+            op = qiskit.optimization.OptimizationProblem("unknown")
+        # If filename does not exist, an exception is raised.
+
+    def test_constructor3(self):
+        # we can pass at most one argument
+        with self.assertRaises(QiskitOptimizationError):
+            op = qiskit.optimization.OptimizationProblem("test", "west")
+
+    def test_constructor_context(self):
+        with qiskit.optimization.OptimizationProblem() as op:
+            op.variables.add(names=['x1', 'x2', 'x3'])
+            self.assertEqual(op.variables.get_num(), 3)
+
+    # def test_end(self):
+    #     op = qiskit.optimization.OptimizationProblem()
+    #     op.end()
+        # TODO: we do not need to release the object
+        # with self.assertRaises(QiskitOptimizationError):
+        #     op.variables.add(names=['x1', 'x2', 'x3'])
+
+    def test_read1(self):
+        op = qiskit.optimization.OptimizationProblem()
+        op.read(self.resource_file)
+        self.assertEqual(op.variables.get_num(), 3)
+
+    def test_write1(self):
+        op = qiskit.optimization.OptimizationProblem()
+        op.variables.add(names=['x1', 'x2', 'x3'])
+        f, fn = tempfile.mkstemp(suffix='.lp')
+        os.close(f)
+        op.write(fn)
+        assert os.path.exists(fn) == 1
+
+    def test_write2(self):
+        op1 = qiskit.optimization.OptimizationProblem()
+        op1.variables.add(names=['x1', 'x2', 'x3'])
+        f, fn = tempfile.mkstemp(suffix='.lp')
```

Comment:
This file should only exist during runtime if I understood the `tempfile` module correctly 🙂

Reply:
https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp
>  the user of mkstemp() is responsible for deleting the temporary file when done with it.
Here is an example where we create a temp file in setup and remove in teardown. The code to try to remove it and ignore any failure is all that you need. Hopefully it does not fail to remove it - it would just lead to more files in the temp folder until such time as they are all deleted.